addons/web/static/src/core/file_upload/file_upload_service.js

```diff
@@ -67,6 +67,18 @@ export const fileUploadService = {
             });
             // Load listener
             xhr.addEventListener("load", () => {
+                if (xhr.readyState === 4 && xhr.status === 400) {
+                    const parser = new DOMParser();
+                    const doc = parser.parseFromString(xhr.responseText, "text/html");
+                    const pTag = doc.querySelector("p");
+                    const errorMessage =
+                        pTag.textContent.trim() || _t("An error occurred while uploading.");
+                    const notificationTime = notificationService.add(errorMessage, {
+                        title: _t("Error"),
+                        sticky: true,
+                    });
+                    setTimeout(notificationTime, 3000);
+                }
```

Comment:
Few things need
1.  return from this block when error triggers ( no need to trigger `FILE_UPLOAD_LOADED`)
2. Good to move ` delete uploads[upload.id];` line to first so it delete from documents
3. Should we user filename in error message?
4. Should we need mark upload state as `error`?